synapseclient/models/mixins/table_components.py

```diff
@@ -4414,13 +4486,56 @@ def csv_to_pandas_df(
     # parse date columns if exists
     if date_columns:
         df = _convert_df_date_cols_to_datetime(df, date_columns)
-    # Turn list columns into lists
+    # Turn list columns into lists and convert items to their proper types
     if list_columns:
         for col in list_columns:
             # Fill NA values with empty lists, it must be a string for json.loads to work
             df.fillna({col: "[]"}, inplace=True)
             df[col] = df[col].apply(json.loads)
 
+            # Convert list items to their proper types based on column type
+            if list_column_types and col in list_column_types:
+                column_type = list_column_types[col]
+                if column_type == "INTEGER_LIST":
+                    # Convert items to int
+                    df[col] = df[col].apply(
+                        lambda x: (
+                            [int(item) for item in x] if isinstance(x, list) else x
```

Comment:
This might not working if the list has NAs. I run into `ValueError: cannot convert float NaN to integer` when the NA is np.nan and `TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NAType'` when the NA is pd.NA.